src/tests/bendersEndToEnd/test_bendersEndToEnd.py

```diff
@@ -0,0 +1,158 @@
+#               value of the variable
+RESULT_FILE_PATH = Path('resultTest.json')
+
+
+def remove_outputs(study_path):
+    output_path = study_path / 'output'
+    for f in Path(output_path).iterdir():
+        if f.is_dir():
+            shutil.rmtree(f)
+
+# launch_optimization : launches optimization part of AntaresXpansion
+# arguments :
+#   - data_path   : Folder containing instance MPS, structure and option files
+#                   Executable will be launched from this folder
+#   - commands    : list of commands to launch
+#                   Example : ['python3', 'test.py', '--arg', 'argValue']
+#   - status      : problem status of current instance
+#                   If this status is not 'OPTIMAL', then the process might return a 
+#                   code different from 0.
+def launch_optimization(data_path, commands, status):
+    
+    # Clean study output
+    # Only optim does not have a specific folder for output
+    # os.remove(data_path + 'out.json')
+    # remove_outputs(data_path)
+
+    # Going to instance folder
+    owd = os.getcwd()
+    os.chdir(data_path)
+
+    # Launching optimization from instance folder
+    process = subprocess.Popen(commands, stdout=subprocess.PIPE, stderr=None)
+    output = process.communicate()
+
+    # Back to working directory
+    os.chdir(owd)
+
+    # Check return value
+    if status == 'OPTIMAL':
+        assert process.returncode == 0
+
+# check_optimization_json_output : Compares values written by optimization in file 
+#                 'out.jons' with expected values, written in resultTest.json.
+# arguments :
+#   - expected_results_dict   : Dict of expected values to compare with ones present in
+#                               in file out.json
+#   - instance_path           : Path to intance folder, to find output file 'out.json'
+def check_optimization_json_output(expected_results_dict, instance_path) :
+
+    # Loading output from optimization process
+    curr_instance_json = {}
+
+    output_path = expected_results_dict['path'] + "/outputs/out.json"
+    with open(output_path, 'r') as jsonFile:
+        curr_instance_json = json.load(jsonFile)
+
+    # Testing optimality
+    prb_status = curr_instance_json['solution']['problem_status']
+    assert prb_status == expected_results_dict['status']
+
+    # Testing optimization output if the instance has a solution
+    if(prb_status == 'OPTIMAL') :
+        value = curr_instance_json['solution']['overall_cost']
+        optimal_variables = curr_instance_json['solution']['values']
+
+        #Testing optimal value
+        np.testing.assert_allclose(value, \
+            expected_results_dict['optimal_value'], rtol=1e-6, atol=0)
+
+        #Testing optmal solution (variables)
+        for var in optimal_variables :
+            np.testing.assert_allclose(optimal_variables[var], \
+                expected_results_dict['optimal_variables'][var], rtol=1e-6, atol=0)
```

Comment:
`np.testing.assert_allclose` also works for vectors/arrays.

Therefore the `for` loop may not be necessary in this case. You may have to convert objects to `np.array` before calling this assertion.